### PR TITLE
Cardano node - hello world.

### PR DESCRIPTION
**Pull request by raduom**
How to test it:
1. (optional?) `test-node $ ./cleanup.sh`
2. `test-node $ ./start-node-01.sh` (in terminal 1)
3. `test-node $ ./start-node-02.sh` (in terminal 2)
4. `tx-inject $ rm -f node-server.sock && ln -s ../test-node/node1/node.sock node-server.sock` (once, or something to that effect in windows - you can also change the path in`config.yaml`)
5. `tx-inject $ ./start-all-servers --without-mock-node` (in terminal 3)

At this point you should start seeing errors on the output in terminal 1.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by j-mueller** [comment]
For the first two steps I had to change the persmissions of `genesis/delegate-keys/delegate1.vrf.skey` and `genesis/delegate-keys/delegate2.vrf.skey` to 600

Where do the scripts and config settings in `test-node` come from? We should add a link / README so that we know where to look when they have to be updated. Or maybe even a git submodule.

File: plutus-pab/app/Command.hs [hunk 1]
```diff
@@ -17,7 +17,7 @@ import           Wallet.Types                            (ContractInstanceId)
 
 -- | A command for which a config.yaml file is required
 data ConfigCommand =
-    MockNode -- ^ Run the mock node service
+    MockNode Bool -- ^ Run the mock node service without starting the server
```
Comment (j-mueller) [inline plutus-pab/app/Command.hs:20]: Should be a datatype, something like `MockServerMode = WithMockServer | WithoutMockServer`

File: plutus-pab/tx-inject/start-all-servers.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-rm -f ./primary.db ./node-server.sock
+rm -f ./primary.db
 cabal build all
```
Comment (j-mueller) [inline plutus-pab/tx-inject/start-all-servers.sh:3]: can we change this to `cabal build plutus-pab`? There are quite a few build targets in `all` that we don't need

**Comment by j-mueller** [comment]
Ok, I finally built all the things and tried it out. Let's merge this after the comments are addressed. (Too bad that it doesn't even complete the handshake though!)

**Comment by raduom** [comment]
> For the first two steps I had to change the permissions of `genesis/delegate-keys/delegate1.vrf.skey` and `genesis/delegate-keys/delegate2.vrf.skey` to 600
> 
> Where do the scripts and config settings in `test-node` come from? We should add a link / README so that we know where to look when they have to be updated. Or maybe even a git submodule.

They come from the `cardano-node` documentation. I could also write the generation as a script, but in that case we loose the static utxo set (which we may want to keep for testing purposes), so I am not sure if we want to do that. 

Because they come from the documentation for the node I am not sure we should add any README (it's in the expected node documentation), but I can do that if you think it would help.

**Comment by raduom** [comment]
> Ok, I finally built all the things and tried it out. Let's merge this after the comments are addressed. (Too bad that it doesn't even complete the handshake though!)

I think one of the reasons why it might not be able to do that is the missing `BlockFetch` protocol which is why I did not bother too much about it.

**Review by j-mueller — APPROVED** [review]

**Pull request merged**